File: C#/Lab7.Library/MathOperations.cs
using System;

namespace Lab7.Library
{
    public class MathOperation 
    {
        public double Logarithm(double value)
        {
            if (value <= 0)
            {
                throw new ArgumentOutOfRangeException("Оскільки логарифм,то аргумент має бути >= 0");
            }
        
        return Math.Log10(value);
        }
    }
}

File: C++/CppLibrary/MathOperations.cpp
#include "MathOperation.h"
#include <cmath>
#include <stdexcept>

double MathOperation::Logarithm(double value)
{
    if (value <= 0)
        throw std::runtime_error("Логарифм: аргумент має бути > 0");

    return std::log10(value);
}